hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/snapshot/OMSnapshotPurgeRequest.java

```diff
@@ -287,18 +219,25 @@ private void updateSnapshotChainAndCache(
       metadataManager.getSnapshotInfoTable().addCacheEntry(
           new CacheKey<>(nextPathSnapInfo.getTableKey()),
           CacheValue.get(trxnLogIndex, nextPathSnapInfo));
-      updatedPathPreviousAndGlobalSnapshots
-          .put(nextPathSnapInfo.getTableKey(), nextPathSnapInfo);
     } else if (nextGlobalSnapInfo != null) {
       nextGlobalSnapInfo.setGlobalPreviousSnapshotId(
           snapInfo.getGlobalPreviousSnapshotId());
       metadataManager.getSnapshotInfoTable().addCacheEntry(
           new CacheKey<>(nextGlobalSnapInfo.getTableKey()),
           CacheValue.get(trxnLogIndex, nextGlobalSnapInfo));
-      updatedPathPreviousAndGlobalSnapshots
-          .put(nextGlobalSnapInfo.getTableKey(), nextGlobalSnapInfo);
     }
 
     snapshotChainManager.deleteSnapshot(snapInfo);
   }
+
+  private SnapshotInfo getUpdatedSnapshotInfo(String snapshotTableKey, OMMetadataManager omMetadataManager)
+      throws IOException {
+    if (updatedSnapshotInfos.containsKey(snapshotTableKey)) {
+      return updatedSnapshotInfos.get(snapshotTableKey);
+    } else {
+      SnapshotInfo snapshotInfo = omMetadataManager.getSnapshotInfoTable().get(snapshotTableKey);
+      updatedSnapshotInfos.put(snapshotTableKey, snapshotInfo);
+      return snapshotInfo;
+    }
```

Comment:
```suggestion
   return updatedSnapshotInfos.computeIfAbsent(snapshotTableKey, k -> omMetadataManager.getSnapshotInfoTable().get(snapshotTableKey));
```

Reply:
I didn't use the lambda function because `IOException` needed to be caught and converted to `RunTimeExcpetion`. And then this `RunTimeException` needs proper handling in `validateAndUpdateCache`.

Reply:
I don't think it matters in this case 1). It is in memory map 2). I doubt the key space is going to be more than 100.

Still, I changed the code to do one look-up to address your concern.